src/beamlime/applications/daemons.py

```diff
@@ -0,0 +1,199 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from abc import ABC
+from typing import Any, List, NewType
+
+from ..constructors import ProviderGroup, SingletonProvider
+from ._parameters import ChunkSize, DataFeedingSpeed
+from ._random_data_providers import RandomEvents
+from ._workflow import Events
+from .base import BaseDaemon, BaseHandler, MessageRouter
+from .handlers import DataReductionHandler, PlotSaver, RawDataSent, StopWatch
+
+
+class DataReductionMessageRouter(MessageRouter):
+    async def run(self) -> None:
+        await super().run()
+        await self.route(
+            StopWatch.Stop(
+                sender=DataReductionMessageRouter,
+                receiver=StopWatch,
+                content=None,
+            )
```

Comment:
The message router is generating messages? That seems odd. Shouldn't it route messages?